dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java

```diff
@@ -851,209 +853,219 @@ else if( fieldbcontentname.startsWith("date") ){
-			Boolean live=con.isLive();
-			searchResult.put("statusIcons", UtilHTML.getStatusIcons(con));
-			searchResult.put("hasLiveVersion", "false");
-			if(!con.isLive() && con.isWorking() && !con.isArchived()){
-				if(APILocator.getVersionableAPI().hasLiveVersion(con)){
-					searchResult.put("hasLiveVersion", "true");
-					searchResult.put("allowUnpublishOfLiveVersion", "true");
-					searchResult.put("inodeOfLiveVersion", APILocator.getVersionableAPI().getContentletVersionInfo(con.getIdentifier(), con.getLanguageId()).getLiveInode());
+				searchResult.put("modUser", user);
+				searchResult.put("owner", con.getOwner());
+				searchResult.put("ownerCanRead", ownerCanRead.toString());
+				searchResult.put("ownerCanWrite", ownerCanWrite.toString());
+				searchResult.put("ownerCanPublish", ownerCanPub.toString());
+				Boolean working = con.isWorking();
+				searchResult.put("working", working.toString());
+				Boolean live = con.isLive();
+				searchResult.put("statusIcons", UtilHTML.getStatusIcons(con));
+				searchResult.put("hasLiveVersion", "false");
+				if (!con.isLive() && con.isWorking() && !con.isArchived()) {
+					if (APILocator.getVersionableAPI().hasLiveVersion(con)) {
+						searchResult.put("hasLiveVersion", "true");
+						searchResult.put("allowUnpublishOfLiveVersion", "true");
+						searchResult.put("inodeOfLiveVersion", APILocator.getVersionableAPI().getContentletVersionInfo(con.getIdentifier(), con.getLanguageId()).getLiveInode());
+					}
 				}
-			}
 
-			searchResult.put("live", live.toString());
-			Boolean isdeleted=con.isArchived();
-			searchResult.put("deleted", isdeleted.toString());
-			Boolean locked=con.isLocked();
-			searchResult.put("locked", locked.toString());
-			searchResult.put("structureInode", con.getStructureInode());
-			searchResult.put("workflowMandatory", String.valueOf(APILocator.getWorkflowAPI().findSchemeForStruct(con.getStructure()).isMandatory()));
-			searchResult.put("contentStructureType", ""+con.getStructure().getStructureType());
+				searchResult.put("live", live.toString());
+				Boolean isdeleted = con.isArchived();
+				searchResult.put("deleted", isdeleted.toString());
+				Boolean locked = con.isLocked();
+				searchResult.put("locked", locked.toString());
+				searchResult.put("structureInode", con.getStructureInode());
+				searchResult.put("workflowMandatory", String.valueOf(APILocator.getWorkflowAPI().findSchemeForStruct(con.getStructure()).isMandatory()));
+				searchResult.put("contentStructureType", "" + con.getStructure().getStructureType());
 
-			// Workflow Actions
+				// Workflow Actions
 
-			List<WorkflowAction> wfActions = new ArrayList<WorkflowAction>();
+				List<WorkflowAction> wfActions = new ArrayList<WorkflowAction>();
 
-			try {
-				wfActions = APILocator.getWorkflowAPI().findAvailableActions( con, currentUser );
-			} catch ( Exception e ) {
-                Logger.error( this, "Could not load workflow actions : ", e );
-            }
+				try {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)